src/System.Private.CoreLib/shared/System/AppContext.cs

```diff
@@ -43,6 +45,8 @@ public static string BaseDirectory
             if (name == null)
                 throw new ArgumentNullException(nameof(name));
 
+            Debug.Assert(s_dataStore != null, "AppContext.Setup should be called first.");
```

Comment:
It would make this `if (s_dataStore == null) return null;` and change `AppContext.SetData` to initialize this using `Interlocked.Exchange`, similar to how the switches dictionary is done. It will keep the shared implementation tree-shaking friendly.

Reply:
Done. Should I keep the asserts as well?